nova/virt/vmwareapi/vmops.py

```diff
@@ -2111,6 +2250,39 @@ def _revert_migration_update_disks(self, vm_ref, instance, vmdk,
     def finish_revert_migration(self, context, instance, network_info,
                                 block_device_info, power_on=True):
         """Finish reverting a resize."""
+        # This DB call can be removed in Train, when the migration object will
+        # be passed as parameter.
+        migration = objects.Migration.get_by_id_and_instance(
+            context, instance.migration_context.migration_id,
+            instance.uuid)
+
+        # Figure out if the revert is being done for a legacy migration
+        # (without a cloned copy of the VM), case when we need to call the
+        # legacy method. This is intended for instances that were in a RESIZED
+        # state before this code change, and users didn't call revert yet.
+        if self._is_legacy_in_place_migration(migration):
+            self._do_finish_revert_in_place_migration(
+                context, instance, block_device_info, network_info)
+        else:
+            self._do_finish_revert_migration(instance, block_device_info,
+                                             network_info)
+
+        if power_on:
+            vm_util.power_on_instance(self._session, instance)
+
+    def _do_finish_revert_migration(self, instance, block_device_info,
+                                    network_info, vm_ref=None):
+        vm_ref = vm_ref or vm_util.get_vm_ref(self._session, instance)
```

Comment:
We should document, that we also call this in the case encountered an error in the migration process. Otherwise, a dev might think from the name, that it's only called on customer revert.